spanner/row.go

```diff
@@ -378,3 +390,174 @@ func (r *Row) ToStructLenient(p interface{}) error {
 		true,
 	)
 }
+
+// SelectAll iterates all rows to the end. After iterating it closes the rows
+// and propagates any errors that could pop up with destination slice partially filled.
+// It expects that destination should be a slice. For each row, it scans data and appends it to the destination slice.
+// SelectAll supports both types of slices: slice of pointers and slice of structs or primitives by value,
+// for example:
+//
+//	type Singer struct {
+//	    ID    string
+//	    Name  string
+//	}
+//
+//	var singersByPtr []*Singer
+//	var singersByValue []Singer
+//
+// Both singersByPtr and singersByValue are valid destinations for SelectAll function.
+//
+// custom setting such as lenient can be passed as an option using DecodeOptions
+// example: to ignore extra columns in the row
+//
+//	var singersByPtr []*Singer
+//	err := spanner.SelectAll(row, &singersByPtr, spanner.WithLenient())
+func SelectAll(rows rowIterator, destination interface{}, options ...DecodeOptions) error {
```

Comment:
Yes we can call this method from outside spanner package.
1) Yes, its intentional at the moment so that no one can use it outside the package, the only reason its present there is to ease in test setup.
2) func SelectAll is exported: Yes and it can be used with RowIterator that is returned for query results 
3) Same as above.
4) Yes we can